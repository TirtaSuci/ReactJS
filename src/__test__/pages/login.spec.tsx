import LoginPage from "@/pages/auth/login";
import { render, screen } from "@testing-library/react";

jest.mock('next/router', () => ({
    useRouter: () => ({
        route: '/auth/login',
        pathname: '/',
        query: {},
        asPath: '/',
        push: jest.fn(),
        replace: jest.fn(),
        reload: jest.fn(),
        back: jest.fn(),
        prefetch: jest.fn().mockResolvedValue(undefined),
        isFallback: false,
        events: {
            on: jest.fn(),
            off: jest.fn(),
            emit: jest.fn(),
        },
    }),
}));

describe("Login Page", () => {
    it("should render correctly", () => {
        const page = render(<LoginPage />);
        expect(page).toMatchSnapshot()
    })
})
