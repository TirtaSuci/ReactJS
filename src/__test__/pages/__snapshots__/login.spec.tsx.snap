// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login"
      >
        <p
          class="login__title"
        >
          Login Page
        </p>
        <div
          class="login__form"
        >
          <div
            class="login__form__background"
          >
            <form
              class="login__form__content"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                class="login__form__content__input"
                id="email"
                name="email"
                required=""
                type="email"
              />
              <label
                for="password"
              >
                Password
              </label>
              <div
                class="passwordWrapper"
              >
                <input
                  class="inputWithToggle"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                />
                <button
                  class="togglePassword"
                  type="button"
                >
                  üëÅÔ∏è
                </button>
              </div>
              <button
                class="login__form__content__button"
                type="submit"
              >
                Login
              </button>
            </form>
            <button
              class="login__form__content__button"
            >
              Login with Google
            </button>
          </div>
          <p
            class="login__form__login"
          >
            Belum punya akun?
             
            <a
              class="login__form__login__link"
              href="/auth/register"
            >
              Register disini
            </a>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login"
    >
      <p
        class="login__title"
      >
        Login Page
      </p>
      <div
        class="login__form"
      >
        <div
          class="login__form__background"
        >
          <form
            class="login__form__content"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              class="login__form__content__input"
              id="email"
              name="email"
              required=""
              type="email"
            />
            <label
              for="password"
            >
              Password
            </label>
            <div
              class="passwordWrapper"
            >
              <input
                class="inputWithToggle"
                id="password"
                name="password"
                required=""
                type="password"
              />
              <button
                class="togglePassword"
                type="button"
              >
                üëÅÔ∏è
              </button>
            </div>
            <button
              class="login__form__content__button"
              type="submit"
            >
              Login
            </button>
          </form>
          <button
            class="login__form__content__button"
          >
            Login with Google
          </button>
        </div>
        <p
          class="login__form__login"
        >
          Belum punya akun?
           
          <a
            class="login__form__login__link"
            href="/auth/register"
          >
            Register disini
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
